src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1269,21 +1272,21 @@ trait RuntimeCosting extends CostingRules with DataCosting with Slicing { IR: Ev
         val uvC = asRep[Costed[GroupElement]](_uv)
         val vvC = asRep[Costed[GroupElement]](_vv)
         val resV: Rep[SigmaProp] = sigmaDslBuilder.proveDHTuple(gvC.value, hvC.value, uvC.value, vvC.value)
-        val cost = opCost(Seq(gvC.cost, hvC.cost, uvC.cost, vvC.cost), costOfDHTuple)
+        val cost = opCost(resV, Seq(gvC.cost, hvC.cost, uvC.cost, vvC.cost), costOfDHTuple)
         RCCostedPrim(resV, cost, mkSizeSigmaProp(gvC.size.dataSize * 4L))
 
       case sigmastate.Exponentiate(In(_l), In(_r)) =>
         val l = asRep[Costed[GroupElement]](_l)
         val r = asRep[Costed[BigInt]](_r)
         val value = l.value.exp(r.value)
-        val cost = opCost(Seq(l.cost, r.cost), costOf(node))
+        val cost = opCost(value, Seq(l.cost, r.cost), costOf(node))
         RCCostedPrim(value, cost, SizeGroupElement)
 
       case sigmastate.MultiplyGroup(In(_l), In(_r)) =>
         val l = asRep[Costed[GroupElement]](_l)
         val r = asRep[Costed[GroupElement]](_r)
         val value = l.value.multiply(r.value)
-        val cost = opCost(Seq(l.cost, r.cost), costOf(node))
+        val cost = opCost(value, Seq(l.cost, r.cost), costOf(node))
```

Comment:
This is also correct, so what goes to RCCostedXXX, should also be used in opCost